docs/DeveloperGuide.md

```diff
@@ -401,6 +400,36 @@ The following sequence diagram shows how the delete patient operation would work
 The following activity diagram summarizes what happens when a user wants to delete an appointment/patient/doctor:
 ![DeleteXYZCommandActivityDiagram](images/DeleteXYZActivityDiagram.png)
 
+### Delete specialisation/medical condition/prescription feature
+This section describes the delete specialisation/medical condition/prescription features.
+
+#### Implementation
+The deletion of a specialisation/medical condition/prescription to MediConnect is facilitated by 'LogicManager'. It extends 'Logic' and stores the mediConnectParser that parses the user input, and the model in which the command is executed. Additionally it implements the following operations:
+
+* `LogicManager#execute()` —  Executes the given user String input and returns a 'CommandResult'
+
+These operations are exposed in the `Ui` interface as `Ui#executeCommand()`.
+
+Given below is an example usage scenario and how the add `Specialisation` mechanism behaves at each step.
+
+Step 1. The user launches the application. The `Database` will be initialized with all data in the order that it was stored in.
+
+Step 2. The user inputs `delete_tag_d 2 Orthopaedic`  to delete an appointment's specialisation into MediConnect.
+The `delete_tag_d` command calls `DeleteSpecialisationtCommandParser#parse` which parses the index argument which is the index of the doctor to delete  
+A new `DeleteSpecialisationtCommand` instance will be created
+
+Step 3. The created `DeleteSpecialisationtCommand` instance is returned to `LogicManager` and its `execute` method is called.
+`DeleteAppointmentCommand#execute` then calls `Model#getFilteredDoctorList` and retrieve the doctor with the given `Index`. 
+Then, the specialisation will be removed from the doctor if exists and replace the existing doctor in Model with the command of `Model#setDoctor`.
+
+The example usage scenario for the delete patient and delete doctor mechanisms would be very similar to the above scenario.
```

Comment:
`delete patient and doctor` change to `delete prescription and medical condition`